### ISSUE TITLE
gather throws error when variable column name not string

### ISSUE THREAD

**Issue by machow** [machow/siuba#180]
```python
import pandas as pd
from siuba import _, gather
df = pd.DataFrame({'id_var': ['x', 'y', 'z'], 0: [1,2,3], 'a': [4,5,6]})

# errors
gather(df, "key", "value", -_.id_var)

# fine
gather(
    df.rename(columns = {0: '0'}),
    "key",
     "value",
    -_.id_var
)
```

expected output:

```
  id_var key  value
0      x   0      1
1      y   0      2
2      z   0      3
3      x   a      4
4      y   a      5
5      z   a      6
```

**Comment by machow** [comment]
Wait, this is an issue with `pd.melt`, which apparently can't accept a column position. It uses np.ravel on the value_vars list. This function is casting the integers to strings....

```
# SAME ERROR
pd.melt(df, id_vars='id_var', value_vars = pd.Index([0, 'a']))
```

**Comment by machow** [comment]
Based on #233, I think this could be resolved by using stack?

**Comment by machow** [comment]
This appears to be resolved in pandas v1.1.
